### PR TITLE
Allow passing `full_config_test`, add weekly testing workflow

### LINKED ISSUE #69: Allow passing full config test option to the test action
It is currently possible to pass it only by an environment variable, when testing locally: https://github.com/canonical/edgex-snap-testing/blob/fcd02aefccc3df92ee26440170f79584ae331d55/test/utils/env.go#L38-L44

### PR DESCRIPTION
**Pull request by MonicaisHer**
resolves #69
testing reschedule event on my local repo: https://github.com/MonicaisHer/edgex-snap-testing/actions/runs/2775182231

### CODE REVIEW

File: .github/workflows/snap-testing.yml [hunk 1]
```diff
@@ -55,9 +60,21 @@ jobs:
...
+        if: ${{ github.event_name == 'pull_request' || github.event_name == 'workflow_dispatch' }}
         uses: ./test
         with:
           name: ${{matrix.name}}
           channel: ${{matrix.channel}}
           platform_channel: ${{matrix.platform_channel}}
           print_logs: ${{matrix.print_logs}}
+
+      - name: Test snap weekly
+        if: ${{ github.event_name == 'schedule' }}
```
Comment (farshidtz) [inline .github/workflows/snap-testing.yml:72]: Was this conditional step really necessary? We could do full config tests for all events.

If you insist on having full config tests only on schedule, then the `with.full_config_test` itself can take an if statement, e.g. `full_config_test: ${{ github.event.name == `schedule` && true ||  false  }}`. It is called the "fake ternary operator".

**Review by farshidtz — CHANGES_REQUESTED** [review]

File: .github/workflows/snap-testing.yml (hunk 1 above)
Comment (MonicaisHer) [reply .github/workflows/snap-testing.yml:72]: Yes, I'm okay with having full config tests for all events. Is a longer testing time for each PRs acceptable?

Comment (farshidtz) [reply .github/workflows/snap-testing.yml:72]: Yes, there will be less PRs as we complete the development of tests. Moreover, we can always change the workflow and run what we need when PR is in draft.

File: .github/workflows/snap-testing.yml [hunk 2]
```diff
@@ -4,6 +4,11 @@ name: Snap Testing
 on:
   pull_request:
     branches: [ main ]
+  schedule:
+    # weekly tests run 6AM every Sunday
+    - cron: "0 3 * * SUN"
+    # TODO: remove this debugging schedule
+    - cron: "50 12 * * MON"
```
Comment (MonicaisHer) [inline .github/workflows/snap-testing.yml:11]: TODO: remove this

**Comment by MonicaisHer** [comment]
Tested reschedule event on my local repo: https://github.com/MonicaisHer/edgex-snap-testing/actions/runs/2775182231

**Comment by farshidtz** [comment]
> Testing reschedule event on my local repo: https://github.com/MonicaisHer/edgex-snap-testing/actions
> 
> It shows `full_config_test=false` (see https://github.com/MonicaisHer/edgex-snap-testing/runs/7612308394?check_suite_focus=true#step:4:8)

We can spend time debugging. But we decided to do the tests always, so why bother?

**Comment by MonicaisHer** [comment]
> > Testing reschedule event on my local repo: https://github.com/MonicaisHer/edgex-snap-testing/actions
> > It shows `full_config_test=false` (see https://github.com/MonicaisHer/edgex-snap-testing/runs/7612308394?check_suite_focus=true#step:4:8)
> 
> We can spend time debugging. But we decided to do the tests always, so why bother?

Make sense. This Pr is ready now.

File: .github/workflows/snap-testing.yml [hunk 3]
```diff
@@ -60,4 +64,5 @@ jobs:
           name: ${{matrix.name}}
           channel: ${{matrix.channel}}
           platform_channel: ${{matrix.platform_channel}}
-          print_logs: ${{matrix.print_logs}}
+          full_config_test: ${{ github.event_name == 'schedule' && true || false }}
```
Comment (farshidtz) [inline .github/workflows/snap-testing.yml:65]:
```suggestion
          full_config_test: true
```
Didn't you agree to enable by default?

File: test/action.yml [hunk 4]
```diff
@@ -24,8 +24,10 @@ inputs:
       This is used when testing services against the edgexfoundry snap.
       This is useful only when 'snap' input is not set.
     required: false
-  print_logs:
-    description: Deprecated. Logs are uploaded as artifacts.
+  full_config_test:
+    description: |
+      Option for enabling full config options test.
+      This is used when testing services' legacy and mixed options.
```
Comment (farshidtz) [inline test/action.yml:30]:
```suggestion
      Option for enabling full config options test to include for e.g. legacy options testing.
```
The actual behavior can change. So better have this as an example.

**Review by farshidtz — CHANGES_REQUESTED** [review]

**Comment by MonicaisHer** [comment]
Failed edgexfoundry test: `LegacyEnvConfig` test and recently newly added `ChangeStartupMsg` test have conflicts.

**Comment by MonicaisHer** [comment]
If we decide on enabling the full config test by default, why don't we enable it on each test suite directly instead of workflow?
```go
TestChangePort: utils.ConfigChangePort{
  TestLegacyEnvConfig:      true,
  TestMixedGlobalAppConfig: false,
}
```

**Comment by farshidtz** [comment]
> If we decide on enabling the full config test by default, why don't we enable it on each test suite directly instead of workflow?
> 
> ```go
> TestChangePort: utils.ConfigChangePort{
>   TestLegacyEnvConfig:      true,
>   TestMixedGlobalAppConfig: true,
> }
> ```

But that will make it enabled for all test runs (incl. manual runs, upstream workflows using the action). We just want to enable it for the snap testing workflow that is internal to this Github project.

**Pull request merged**